hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -541,6 +558,9 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
       checkpointStr = dataAndCheckpoint.getCheckpointForNextBatch();
       boolean reconcileSchema = props.getBoolean(DataSourceWriteOptions.RECONCILE_SCHEMA().key());
       if (this.userProvidedSchemaProvider != null && this.userProvidedSchemaProvider.getTargetSchema() != null) {
+        if (rowBulkInsert) {
+          return new InputBatch(transformed, checkpointStr, this.userProvidedSchemaProvider);
```

Comment:
I think we need to handle quarantine before returning here

Reply:
yes. main purpose of using row writer is for perf. but if we were to do this, might incur perf hit. So, in V0, I am punting error table for row writer use-cases.